goshopify_test.go

```diff
@@ -378,6 +347,96 @@ func TestDo(t *testing.T) {
 	}
 }
 
+func TestRateLimitRetry(t *testing.T) {
```

Comment:
I think you could combine this test with `TestServiceUnavailableRetry` and expand it to better cover the scenarios 1) request immediately succeeds 2) requests fail and are retried (plus a check that they're retried the right number of times) and 3) reuqests are retried and go over the max and it fails.
Look into using table driven tests https://github.com/golang/go/wiki/TableDrivenTests and make Reponder response, responder # of 429s, client retries, expected outcome (error or not), etc driven by the test table.

Reply:
refactored! thanks for the feedback